From 94d059e9f1fb26bdd6cd1ad2a4fe0d1f41838f2a Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Thu, 20 Aug 2020 23:30:02 -0300
Subject: [PATCH 034/322] [FIO extras] cmd: bootm: allow fit with imx_hab

IMX_HAB assumes that the only authentication method supported is via
HAB, which is now what we always want necessarily, specially since FIT
signature checks is also another valid secure boot method.

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
Signed-off-by: Vanessa Maegima <vaness.maegima@foundries.io>
---
 cmd/bootm.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/cmd/bootm.c b/cmd/bootm.c
index 4b6209a0bf1..bdd44fcc31e 100644
--- a/cmd/bootm.c
+++ b/cmd/bootm.c
@@ -181,15 +181,9 @@ int do_bootm(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 		/* Do this authentication in boota command */
 		break;
 #endif
-#ifdef CONFIG_FIT
+#if defined(CONFIG_FIT) && defined(CONFIG_FIT_SIGNATURE)
 	case IMAGE_FORMAT_FIT:
-		buf = (void *)map_sysmem(addr, 0);
-		size_t size = fit_get_size(buf);
-		unmap_sysmem(buf);
-		if (authenticate_image(addr, size) != 0) {
-			printf("Authenticate FIT image Fail, Please check\n");
-			return 1;
-		}
+		/* Assume authentication is done by fit signature */
 		break;
 #endif
 	default:
-- 
2.34.1

